### PR TITLE
Add missing tests

### PR DESCRIPTION
**Pull request by kunicmarko20**
I am targeting this branch, because this is BC.

## Subject

Adding tests that are missing, as much as I can.

### CODE REVIEW

**Comment by kunicmarko20** [comment]
Should probably move to prophesize (https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/807/files#diff-6dfd8ead8e135a17ae2f1cec987ee09cR55) :sweat_smile:

**Comment by greg0ire** [comment]
This is great, but pedantic since there is zero benefits when upgrading. The benefit come when developing :) Removed the changelog. prophesize is allowed and recommended for new tests, so if you want to change, you can.

**Comment by jordisala1991** [comment]
Yes, if you can move to prophecy that would be great

File: tests/Block/AuditBlockServiceTest.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+     * @group legacy
+     */
+    public function testExecute()
+    {
+        $block = $this->prophesize(Block::class);
+        $revision = $this->prophesize(Revision::class);
+        $blockContext = $this->prophesize(BlockContext::class);
+
+        $blockContext->getBlock()
+            ->willReturn($block->reveal())
```
Comment (jordisala1991) [inline tests/Block/AuditBlockServiceTest.php]: this can go one lined and won't be more than 100 chars

File: tests/SonataDoctrineORMAdminBundleTest.php [hunk 2]
```diff
@@ -25,17 +26,21 @@ class SonataDoctrineORMAdminBundleTest extends TestCase
...
-        $containerBuilder->expects($this->exactly(3))
-            ->method('addCompilerPass')
-            ->withConsecutive(
-                [$this->isInstanceOf(AddGuesserCompilerPass::class)],
-                [$this->isInstanceOf(AddTemplatesCompilerPass::class)],
-                [$this->isInstanceOf(AddAuditEntityCompilerPass::class)]
-            );
+        $containerBuilder
+            ->addCompilerPass(Argument::type(AddGuesserCompilerPass::class))
+            ->shouldBeCalledTimes(1);
```
Comment (jordisala1991) [inline tests/SonataDoctrineORMAdminBundleTest.php]: All this can be simplified to shouldBeCalled().
And could be done in one line probably (have to check chars)

Comment (kunicmarko20) [reply tests/SonataDoctrineORMAdminBundleTest.php]: Well, I can do it in one line and check for `CompilerPassInterface` instead of specific class.

Comment (jordisala1991) [reply tests/SonataDoctrineORMAdminBundleTest.php]: That's another option. There are certain things that can be done with phpunit mocks that are not direct translated to how you do it with prophecy, but there is always a way in which you get a cleaner code. atleast IMO.

File: tests/Block/AuditBlockServiceTest.php [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+ */
+class AuditBlockServiceTest extends AbstractBlockServiceTestCase
+{
+    private $simpleThingsAuditReader;
+    private $blockService;
+
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->simpleThingsAuditReader = $this->prophesize(SimpleThingsAuditReader::class);
```
Comment (greg0ire) [inline tests/Block/AuditBlockServiceTest.php]: Can you avoid mocking this and use the real thing instead or is it too cumbersome?

Comment (kunicmarko20) [reply tests/Block/AuditBlockServiceTest.php]: Then I would need to mock things inside of `SimpleThingsAuditReader`?

Comment (greg0ire) [reply tests/Block/AuditBlockServiceTest.php]: I don't know what dependencies it has, that's why I'm asking. If there are too many, just don't do it.

Comment (kunicmarko20) [reply tests/Block/AuditBlockServiceTest.php]:
```     */
    public function __construct(EntityManagerInterface $em, AuditConfiguration $config, MetadataFactory $factory)
    {
        $this->em = $em;
        $this->config = $config;
        $this->metadataFactory = $factory;
        $this->platform = $this->em->getConnection()->getDatabasePlatform();
        $this->quoteStrategy = $this->em->getConfiguration()->getQuoteStrategy();
    }
```

Too many :sweat_smile: , wouldn't that basically be testing 3rd party code?

Comment (greg0ire) [reply tests/Block/AuditBlockServiceTest.php]: Yeah drop it for that particular case, it's way too complicated. The goal is to avoid mocking internal calls, catch `trigger_error` and deprecations if any

File: tests/Builder/ShowBuilderTest.php [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    }
+
+    public function testAddFieldNoType()
+    {
+        $fieldDescriptionCollection = $this->prophesize(FieldDescriptionCollection::class);
+        $fieldDescription = $this->prophesize(FieldDescriptionInterface::class);
+        $admin = $this->prophesize(AdminInterface::class);
+        $typeGuess = $this->prophesize(TypeGuess::class);
+        $modelManager = $this->prophesize(ModelManager::class);
+        $typeGuess = $this->prophesize(TypeGuess::class);
```
Comment (jordisala1991) [inline tests/Builder/ShowBuilderTest.php]: duplicated line with 60

Comment (kunicmarko20) [reply tests/Builder/ShowBuilderTest.php]: nice catch!

File: tests/Block/AuditBlockServiceTest.php [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        );
+    }
+
+    /**
+     * @group legacy
+     */
+    public function testExecute()
+    {
+        $block = $this->prophesize(Block::class);
+        $revision = $this->prophesize(Revision::class);
```
Comment (greg0ire) [inline tests/Block/AuditBlockServiceTest.php]: This one you can easily avoid mocking

Comment (greg0ire) [reply tests/Block/AuditBlockServiceTest.php]: Both, in fact.

Comment (kunicmarko20) [reply tests/Block/AuditBlockServiceTest.php]: even if  I assert them down?

File: tests/Builder/ShowBuilderTest.php (hunk 4 above)
Comment (jordisala1991) [reply tests/Builder/ShowBuilderTest.php]: I am looking at what scrutinizer says, there are a few errors out there, can you check it?

Comment (kunicmarko20) [reply tests/Builder/ShowBuilderTest.php]: errors on `AuditReaderTest` do not make sense?

File: tests/Guesser/FilterTypeGuesserTest.php [hunk 6]
```diff
@@ -43,4 +50,162 @@ public function testThrowsOnMissingField()
...
+            ],
+            [
+                'smallint',
+                $numberType,
+                Guess::MEDIUM_CONFIDENCE,
+                NumberType::class
+            ],
+            [
+                'string',
+                $strngType = 'doctrine_orm_string',
```
Comment (OskarStark) [inline tests/Guesser/FilterTypeGuesserTest.php]:
```diff
- $stringType
+ $stringType
```

File: tests/Guesser/FilterTypeGuesserTest.php [hunk 7]
```diff
@@ -43,4 +50,162 @@ public function testThrowsOnMissingField()
...
+                Guess::MEDIUM_CONFIDENCE,
+                TextType::class
+            ],
+            [
+                'text',
+                $strngType,
```
Comment (OskarStark) [inline tests/Guesser/FilterTypeGuesserTest.php]:
```diff
- $stringType
+ $stringType
```

File: tests/Guesser/FilterTypeGuesserTest.php [hunk 8]
```diff
@@ -43,4 +50,162 @@ public function testThrowsOnMissingField()
...
+                TextType::class
+            ],
+            [
+                'time',
+                'doctrine_orm_time',
+                Guess::HIGH_CONFIDENCE,
+            ],
+            [
+                'somefake',
+                $strngType,
```
Comment (OskarStark) [inline tests/Guesser/FilterTypeGuesserTest.php]:
```diff
- $stringType
+ $stringType
```

File: tests/Builder/ShowBuilderTest.php (hunk 4 above)
Comment (jordisala1991) [reply tests/Builder/ShowBuilderTest.php]: It is because of docblocks declaring types

File: tests/Builder/ShowBuilderTest.php [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+use Sonata\DoctrineORMAdminBundle\Model\ModelManager;
+use Symfony\Component\Form\Guess\TypeGuess;
+
+/**
+ * @author Marko Kunic <kunicmarko20@gmail.com>
+ */
+class ShowBuilderTest extends TestCase
+{
+    /**
+     * @var TypeGuesserInterface
```
Comment (jordisala1991) [inline tests/Builder/ShowBuilderTest.php]: Adding this makes scrutinizer think this is a TypeGuesserInterface, but it is a ObjectProphecy (or something like that). IMO I would remove all those comments from test classes, because they do not give a lot of information, and just make noise

File: tests/Guesser/FilterTypeGuesserTest.php (hunk 6 above)
Comment (kunicmarko20) [reply tests/Guesser/FilterTypeGuesserTest.php]: thank you, fixed.

File: tests/Builder/ShowBuilderTest.php (hunk 9 above)
Comment (kunicmarko20) [reply tests/Builder/ShowBuilderTest.php]: my bad, fixed.

File: tests/Block/AuditBlockServiceTest.php (hunk 5 above)
Comment (greg0ire) [reply tests/Block/AuditBlockServiceTest.php]: For `$revision` it will simplify things, and for `$block`, you did not write any assertions. The only thing lost is the number of times `getRev` is called, but that's an implementation detail, and we only care about the outcome of the test, right?

File: tests/Builder/ShowBuilderTest.php [hunk 10]
```diff
@@ -175,13 +178,35 @@ public function testFixFieldDescription()
...
-        $fieldDescription->setTemplate('@SonataAdmin/CRUD/Association/show_one_to_one.html.twig')
+        $fieldDescription->setTemplate($template)
             ->shouldBeCalledTimes(1);
 
         $admin->attachAdminClass($fieldDescription)->shouldBeCalledTimes(1);
 
         $this->showBuilder->fixFieldDescription($admin->reveal(), $fieldDescription->reveal());
     }
+
+    public function testFixFieldDescriptionData()
```
Comment (greg0ire) [inline tests/Builder/ShowBuilderTest.php]: A data provide that starts with `test`? Is phpunit ok with this?

File: tests/Block/AuditBlockServiceTest.php [hunk 11]
```diff
@@ -43,19 +43,12 @@ protected function setUp()
      */
     public function testExecute()
     {
-        $block = $this->prophesize(Block::class);
-        $revision = $this->prophesize(Revision::class);
```
Comment (greg0ire) [inline tests/Block/AuditBlockServiceTest.php]: I'd thought you were going to do `$revision = new Revision()`, why did you remove it?

File: tests/Builder/ShowBuilderTest.php (hunk 10 above)
Comment (kunicmarko20) [reply tests/Builder/ShowBuilderTest.php]: I did not have any errors, so it should be ok but I will rename it as this was a mistake.

File: tests/Block/AuditBlockServiceTest.php (hunk 11 above)
Comment (kunicmarko20) [reply tests/Block/AuditBlockServiceTest.php]: My bad, I understood `avoid mocking` as removing, will be fixed.

**Comment by greg0ire** [comment]
> +17.6%

That's impressive!

**Comment by kunicmarko20** [comment]
My goal is at least `70%` coverage, probably today or tomorrow 😄

**Comment by kunicmarko20** [comment]
I guess it is done if you have time review it. Should I squash commits? also, any idea what to do about the test that is failing on prefer lowest?

**Comment by dmarkowicz** [comment]
Lowest version passing test is:
`"simplethings/entity-audit-bundle": "0.5 - 0.9 || ^1.0",`
What about simplify it to:
`"simplethings/entity-audit-bundle": "^0.9 || ^1.0",`

File: tests/Builder/DatagridBuilderTest.php [hunk 12]
```diff
@@ -11,69 +11,159 @@
...
+        return [
+            [
+                Pager::TYPE_SIMPLE,
+                SimplePager::class,
+            ],
+            [
+                Pager::TYPE_DEFAULT,
+                Pager::class,
+            ],
+        ];
```
Comment (greg0ire) [inline tests/Builder/DatagridBuilderTest.php]: It's good practice to specify a description for each item of a data provider as its key
```
return [
   'simple' => [
…
]
]
```

Comment (kunicmarko20) [reply tests/Builder/DatagridBuilderTest.php]: will add keys, any idea about the issue with the test, what @dmarkowicz suggested make sense but would that be a BC break if we move to 0.9?

Comment (jordisala1991) [reply tests/Builder/DatagridBuilderTest.php]: Increase dependencies is not a BC break, IMO let's do it

File: tests/Filter/DateTimeFilterTest.php [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+ * @author Marko Kunic <kunicmarko20@gmail.com>
+ */
+class DateTimeFilterTest extends TestCase
+{
+    public function testEmpty()
+    {
+        $filter = new DateTimeFilter();
+        $filter->initialize('field_name', ['field_options' => ['class' => 'FooBar']]);
+
+        $builder = new ProxyQuery(new QueryBuilder());
```
Comment (jordisala1991) [inline tests/Filter/DateTimeFilterTest.php]: This queryBuilder should have a `use` right?

File: tests/Filter/DateTimeRangeFilterTest.php [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+ * @author Marko Kunic <kunicmarko20@gmail.com>
+ */
+class DateTimeRangeFilterTest extends TestCase
+{
+    public function testEmpty()
+    {
+        $filter = new DateTimeRangeFilter();
+        $filter->initialize('field_name', ['field_options' => ['class' => 'FooBar']]);
+
+        $builder = new ProxyQuery(new QueryBuilder());
```
Comment (jordisala1991) [inline tests/Filter/DateTimeRangeFilterTest.php]: same here

File: tests/Filter/TimeFilterTest.php [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+ * @author Marko Kunic <kunicmarko20@gmail.com>
+ */
+class TimeFilterTest extends TestCase
+{
+    public function testEmpty()
+    {
+        $filter = new TimeFilter();
+        $filter->initialize('field_name', ['field_options' => ['class' => 'FooBar']]);
+
+        $builder = new ProxyQuery(new QueryBuilder());
```
Comment (jordisala1991) [inline tests/Filter/TimeFilterTest.php]: and here

File: tests/Filter/DateTimeFilterTest.php (hunk 13 above)
Comment (kunicmarko20) [reply tests/Filter/DateTimeFilterTest.php]: There is a QueryBuilder class i that folder https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/3.x/tests/Filter/QueryBuilder.php

**Comment by kunicmarko20** [comment]
Test are finally passing, could you review this and lets merge it :smile:

**Comment by jordisala1991** [comment]
Great work BTW

**Comment by kunicmarko20** [comment]
Thank you, I am happy with +26%, we are over 70% on this bundle :tada:

**Comment by kunicmarko20** [comment]
Also started to like prophecy a lot, I think I can't (and don't want to) go back to `createMock` way :smile:

**Comment by jordisala1991** [comment]
Once you try, you never go back ;)

Those tests could probably be ported to MongoDBAdminBundle and PHPCRAdminBundle without much trouble. We should probably give it a try some day :)

**Comment by kunicmarko20** [comment]
You are right, some of the tests could be ported, will look into it.

**Comment by jordisala1991** [comment]
Thank you @kunicmarko20
